### PR TITLE
Crawler: retry with new query when history ends

### LINKED ISSUE #78: Partly missing tweet history
For most of our main characters, the tweet history goes back all the way until 2011-2010. However, I noticed that for some of them we only have data from mid / end 2014 on (Daenerys Targaryen, Sansa Stark) or even later (Tyrion Lannister starts in August 2015), even though those characters were already important before. The history also looks cut off unnaturally. What could prevent the crawler from getting those older tweets?

### PR DESCRIPTION
**Pull request by julienschmidt**
Fixes #78 

However I'm still waiting for the crawler to finish crawling Tweets mentioning Jon Snow. We might have a new problem now :laughing:

### CODE REVIEW

**Comment by julienschmidt** [comment]
ou ou. The crawler just hang without any error again :worried:

**Comment by julienschmidt** [comment]
Sorry, my fault. I forgot the `.catch(debug.error)` in my test function.

**Comment by marcusnovotny** [comment]
In other words: Ready to merge? :)

**Comment by julienschmidt** [comment]
:+1: 

**review the code** and then you may merge it :wink:

File: crawler/mobile.js [hunk 1]
```diff
@@ -145,22 +145,39 @@ mobile.crawl = function(character, full) {
     const nextRe    = /search\?q=(.+)"> Load older Tweets/;
     const doneRe    = /<div class="noresults"> No results for/;
 
+    // Number of Tweets after which retry with another "until:(lastDay)" query
+    // when the history ends. For some popular characters the history ends early.
+    const retryThreshold = 5000;
```
Comment (marcusnovotny) [inline crawler/mobile.js:150]: Just curious, why use 5000 tweets as the threshold?

**Comment by marcusnovotny** [comment]
Looks fancy as always! :+1:

File: crawler/mobile.js (hunk 1 above)
Comment (julienschmidt) [inline crawler/mobile.js:150]: Well.. we have to start somewhere :P
I have a hunch that Twitter tries to keep a history of roughly 10k tweets for popular searches. But I actually have no idea. So I just put a random number large enough to not retry every time.

**Comment by julienschmidt** [comment]
Yup we have a new problem now:

```
﻿﻿{ [MongoError: errmsg: "Sort operation used more than the maximum 33554432 bytes of RAM. Add an index, or specify a smaller limit."]
  name: 'MongoError',
  message: 'errmsg: "Sort operation used more than the maximum 33554432 bytes of RAM. Add an index, or specify a smaller limit."',
  waitedMS: 0,
  ok: 0,
  errmsg: 'errmsg: "Sort operation used more than the maximum 33554432 bytes of RAM. Add an index, or specify a smaller limit."',
  code: 96 }
```

Let's start thinking about adding additional indices and compression :laughing:

**Comment by julienschmidt** [comment]
At least on my laptop `block_compressor=snappy` is used by default. So the data is already compressed...

**Comment by julienschmidt** [comment]
We have really serious problems now...

```
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
[1]    34838 abort      node app
```

**Comment by marcusnovotny** [comment]
Mine still seems to run just fine, where do you think is that coming from?

**Comment by julienschmidt** [comment]
I have some more fixes (see #88) and crawled Jon Snow last night - who is mentioned about ~800.000 times

**Pull request merged**